libs/telemetry/logger.go

```diff
@@ -0,0 +1,150 @@
+package telemetry
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/databricks/cli/libs/log"
+	"github.com/databricks/cli/libs/telemetry/protos"
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/client"
+	"github.com/databricks/databricks-sdk-go/config"
+	"github.com/google/uuid"
+)
+
+// Environment variable to disable telemetry. If this is set to any value, telemetry
+// will be disabled.
+const DisableEnvVar = "DATABRICKS_CLI_DISABLE_TELEMETRY"
+
+func Log(ctx context.Context, event protos.DatabricksCliLog) {
+	fromContext(ctx).log(event)
+}
+
+func SetExecutionContext(ctx context.Context, ec protos.ExecutionContext) {
+	fromContext(ctx).setExecutionContext(ec)
+}
+
+func HasLogs(ctx context.Context) bool {
```

Comment:
It could be worth having this one-line wrappers for operations that are scattered throughout the code, but HasLogs and SetExecutionContext seems like they are going to be used once.

Reply:
Inlined `SetExecutionContext. We still need `HasLogs` since it's used outside the telemetry package.